### PR TITLE
Feature/acczi landing detector

### PR DESCRIPTION
**Pull request by magate**

### CODE REVIEW

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -2950,6 +2950,8 @@ void QuadPlane::takeoff_controller(void)
     plane.nav_roll_cd = 0;
     plane.nav_pitch_cd = 0;
 
+    landing_detector->enabled();
```
Comment (IamPete1) [inline ArduPlane/quadplane.cpp:2953]:
```suggestion
    plane.landing_detector.enabled();
```
You don't fill in the local pointer, so you have to go via plane.

File: ArduPlane/quadplane.h [hunk 2]
```diff
@@ -205,6 +207,7 @@ class QuadPlane
     AC_PosControl *pos_control;
     AC_WPNav *wp_nav;
     AC_Loiter *loiter_nav;
+    AP_LandingDetector *landing_detector;
```
Comment (IamPete1) [inline ArduPlane/quadplane.h:210]: This is never filled in. The others here are created by quadplane. The other way is the same as AHRS where you take a reference and init it in the quadplane constructor.

File: ArduPlane/quadplane.h [hunk 3]
```diff
@@ -59,6 +60,7 @@ class QuadPlane
     friend class ModeQAutotune;
     friend class ModeQAcro;
     friend class ModeLoiterAltQLand;
+    friend class AP_LandingDetector;
```
Comment (IamPete1) [inline ArduPlane/quadplane.h:63]: Should not need this.

File: ArduPlane/quadplane.cpp (hunk 1 above)
Comment (magate) [reply ArduPlane/quadplane.cpp:2953]: Thanks. It was `plane.g2.landing_detector.enabled()`;. Slowly learning the C++ architecture.

**Comment by khancyr** [comment]
I surely miss something but this does nothing than adding a single parameter that is unused ?

**Comment by magate** [comment]
> I surely miss something but this does nothing than adding a single parameter that is unused ?
 
Apologies @khancyr . Needed some input from Peter on resolving a segfault on my way to feature development. Will close this PR to avoid anymore confusion.

**Comment by khancyr** [comment]
no problem with that (even if we shouldn't trigger our CI much for testing as it is costly). Best put a message in the description so other can understand what is happening ! You can also share a link to your personnal github , it should be the same

**Pull request closed without merging**